### PR TITLE
Harvest: Update documentation and comments

### PR DESCRIPTION
**Pull request by gregorylegarec**
Also two minor refactors in the first two commits.

### CODE REVIEW

File: packages/cozy-harvest-lib/README.md [hunk 1]
```diff
@@ -1,107 +1,55 @@
-Cozy-Harvest-Lib
-=====
+# Cozy-Harvest-Lib
+
+# Target
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/README.md]: Goal ?

Comment (gregorylegarec) [reply packages/cozy-harvest-lib/README.md]: I think I can rename Target 😅

File: packages/cozy-harvest-lib/src/components/AccountForm/AccountField.jsx [hunk 2]
```diff
@@ -11,10 +11,18 @@ import {
   ROLE_IDENTIFIER
 } from '../../helpers/manifest'
 
+/**
+ * AccountField encapsulate an unique Cozy-UI Field component.
+ * It maps its own props to Cozy-UI Field ones, and deal with i18n and locale
+ * logic.
+ *
+ * AccountField's props should be coming from konnector manifest
+ * fields property. We expect this object to have been sanitized.
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/AccountForm/AccountField.jsx]: By this object you mean `props.fields` ?

File: packages/cozy-harvest-lib/src/components/AccountForm/index.jsx [hunk 3]
```diff
@@ -141,11 +168,45 @@ export class AccountForm extends PureComponent {
...
+   * Indicates if the AccountForm has to show errors. Sometimes errors may be
+   * displayed elsewhere. However, a KonnectorJobError corresponding to a login
+   * error is always displayed.
+   * @type {Boolean}
+   */
   showError: PropTypes.bool,
-  submitting: PropTypes.bool
+  /**
+   * Indicates if the form should be rendered as submitting data or busy.
+   * Tipically updated after an `onSubmit` call.
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/AccountForm/index.jsx]: Typically

File: packages/cozy-harvest-lib/src/components/AccountForm/AccountFields.jsx [hunk 4]
```diff
@@ -51,10 +57,31 @@ export class AccountFields extends PureComponent {
...
+   */
   container: PropTypes.node,
+  /**
+   * Indicates if all the fields are disabled
+   * @type {Boolean}
+   */
   disabled: PropTypes.bool,
+  /**
+   * The sanitized konnector.fields object used do define the form fields
+   * @type {[type]}
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/AccountForm/AccountFields.jsx]: @type {object}

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx [hunk 5]
```diff
@@ -183,23 +183,98 @@ export class TriggerManager extends Component {
...
+  /**
+   * Callback invoked when the trigger has been launched and that the login
+   * to the remote service has succeed.
+   * @type {Func}
+   */
   onLoginSuccess: PropTypes.func,
-  onSuccess: PropTypes.func
+  /**
+   * Callback invoked when the trigger has been launched and the job ended.
+   * Can be called before `onLoginSucces` for some konnectors.
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: "onLoginSuccess". I don't understand this comment. Is it something I must pay attention to ?

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx [hunk 6]
```diff
@@ -183,23 +183,98 @@ export class TriggerManager extends Component {
...
+   * Job mutations
+   * @type {Func}
+   */
   waitForLoginSuccess: PropTypes.func.isRequired,
-  // hooks
+  //
+  // Callbacks
+  //
+  /**
+   * Callback invoked when the trigger has been launched and that the login
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: "Callback invoked when the trigger has been launched and the login to the remote service has succeeded"

File: packages/cozy-harvest-lib/src/components/TriggerManager.jsx [hunk 7]
```diff
@@ -183,23 +183,98 @@ export class TriggerManager extends Component {
...
+   */
   saveAccount: PropTypes.func.isRequired,
+  /**
+   * Trigger mutations
+   * @type {Func}
+   */
   statDirectoryByPath: PropTypes.func,
+  /**
+   * Job mutations
+   * @type {Func}
```
Comment (ptbrowne) [inline packages/cozy-harvest-lib/src/components/TriggerManager.jsx]: Func or Function ?

**Pull request merged**